lib/puppet/provider/user/aix.rb

```diff
@@ -315,4 +315,19 @@ def create
   #    `inclusive`, `minimum`.
   # UNSUPPORTED
 
+  # Deletes this instance resource
+  def delete
+    execute(deletecmd)
+
+    if @resource.managehome? && Dir.exist?(Dir.home(@resource[:name]))
+      FileUtils.remove_entry_secure(Dir.home(@resource[:name]), true)
+    end
+
+    # Recollect the object info so that our current properties reflect
+    # the actual state of the system. Otherwise, puppet resource reports
+    # the wrong info. at the end. Note that this should return nil.
+    object_info(true)
+  rescue Puppet::ExecutionFailure => detail
+    raise Puppet::Error, _("Could not delete %{resource} %{name}: %{detail}") % { resource: @resource.class.name, name: @resource.name, detail: detail }, detail.backtrace
+  end
```

Comment:
other solution will be to directly call the  `super` method and just handle the home directory here.
should this work if the home directory is not specified on the resource (and use default from Puppet::Etc?)

Reply:
Are we sure `Dir.home` will be correct, and for example not `/`? Did `userdel -r` not work?

Reply:
Removing `/` sounds really dangerous, for sure we should protect against.

We are not sure what are the differences between `rmuser` and `userdel`, but we supposed there was a reason for which we are not using `useradd` provider on aix (NIS compatilibity?). One difference is that rmuser is not removing user from `passwd` while `deluser` does.

On the other side, this is the `delete` operation that will remove the used, the risk of breaking anything is low.

Reply:
At a first glance, `userdel` seemed like a good option (since it is available by default on [AIX 7.1](https://www.ibm.com/docs/en/aix/7.1?topic=u-userdel-command) and [AIX 7.2](https://www.ibm.com/docs/en/aix/7.2?topic=u-userdel-command) and it has the `-r` cli option to take care of the home directory itself) but after a bit of digging through our code, we found out that our [user delete code](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/provider/aix_object.rb#L348-L350) for AIX is using [ia module arguments](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/provider/aix_object.rb#L332-L337), including the more impactful `-R` which allows Puppet to manage users created from alternate Identification and Authentication mechanisms such as `LDAP` so we sticked with this less impactful solution.